src/paginator.js

```diff
@@ -148,13 +149,21 @@ paginator.parseArguments_ = function(args) {
     }
   }
 
-  return {
+  var parsedArguments = {
     query: query || {},
     autoPaginate: autoPaginate,
     maxApiCalls: maxApiCalls,
     maxResults: maxResults,
     callback: callback,
   };
+
+  parsedArguments.streamOptions = extend(true, {}, parsedArguments.query);
+  delete parsedArguments.streamOptions.autoPaginate;
+  delete parsedArguments.streamOptions.maxApiCalls;
+  delete parsedArguments.streamOptions.maxResults;
+  delete parsedArguments.streamOptions.pageSize;
```

Comment:
`autoPaginate`, `maxApiCalls`, etc. seem to be properties of `parsedArguments` itself, rather than `parsedArguments.query`. So, if `parsedArguments.streamOptions` comes from copying `parsedArguments.query`, wouldn't those properties be absent from `parsedArguments.streamOptions` and thus not need to be deleted?

Other than that, this LGTM!

Reply:
That's true, however, they originate from `query`. After your comment though, I did find one thing we don't need to delete, "maxApiCalls", as that is deleted in earlier code.

Reply:
Good to know!